src/main/java/org/takes/http/BkBasic.java

```diff
@@ -116,4 +120,32 @@ private static Response failure(final Throwable err) {
         );
     }
 
+    /**
+     * Adds custom headers with information about socket.
+     * @param req Request
+     * @param socket Socket
+     * @return Request with custom headers
+     */
+    private static Request addSocketHeaders(final Request req,
+        final Socket socket) {
+        Request mreq = req;
+        mreq = new RqWithHeader(
```

Comment:
@ikhvostenkov No need for `HashMap`. Let's keep the API immutable. How about adding a method like this:

```java
public RqWithHeader withAdditionalHeader(CharSequence name, CharSequence value) {
    return new RqWithHeader(this, name, value);
}
```

This should enable the fluent interface that I suggested.